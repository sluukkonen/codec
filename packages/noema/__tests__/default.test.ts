import { defaultTo, string } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should return the default if decoding undefined", () => {
  expectParseSuccess(defaultTo(string, "abc"), undefined, "abc")
})

test("should defer to the underlying codec if the value is not undefined", () => {
  expectParseSuccess(defaultTo(string, "abc"), "foo")
  expectParseFailure(defaultTo(string, "abc"), null)
})
