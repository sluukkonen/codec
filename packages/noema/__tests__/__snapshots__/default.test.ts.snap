// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should defer to the underlying codec if the value is not undefined 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "path": [],
    "received": "null",
  },
]
`;
